lib/src/mock_document_reference.dart

```diff
@@ -217,7 +217,9 @@ class MockDocumentReference<T extends Object?> implements DocumentReference<T> {
     rootParent.remove(id);
     _firestore.removeSavedDocument(path);
     QuerySnapshotStreamManager().fireSnapshotUpdate(firestore, path);
-    fireSnapshotUpdate();
+    if (_exists()) {      
```

Comment:
On line 218, since we remove the saved document, I believe it won't exist. So perhaps we can just remove the fireSnapshotUpdate altogether?

Reply:
Sorry, I missed this. Thanks for pointing that out and doing the change. 💯